server/utils.js

```diff
@@ -52,7 +52,33 @@ var convertDirectionsToStreet = function(req, callback) {
       }, 20 * i);
     })(i);
   }
-}
+};
+
+
+var findCrimesByLine = function(directions, callback) {
+  var asyncNumCrimes = directions.map((street) => {
+    return new Promise((res, rej) => {
+      dbCrime.findCrimesByLine(street, function(err, crimes) {
+        if (err) {
+          console.error(err);
+          rej(err);
+        } else {
+          res(crimes.length);
+        }
+      });
+    });
+  });
+  Promise.all(asyncNumCrimes).then(callback);
```

Comment:
should this be
Promise.all(asyncNumCrimes).then( (values) => {
  callback(values);
};
?